src/neo/Wallets/NEP6/NEP6Wallet.cs

```diff
@@ -49,10 +48,12 @@ public class NEP6Wallet : Wallet
         /// Loads or creates a wallet at the specified path.
         /// </summary>
         /// <param name="path">The path of the wallet file.</param>
+        /// <param name="password">The password of the wallet.</param>
         /// <param name="settings">The <see cref="ProtocolSettings"/> to be used by the wallet.</param>
         /// <param name="name">The name of the wallet. If the wallet is loaded from an existing file, this parameter is ignored.</param>
-        public NEP6Wallet(string path, ProtocolSettings settings, string name = null) : base(path, settings)
+        public NEP6Wallet(string path, string password, ProtocolSettings settings, string name = null) : base(path, settings)
```

Comment:
Since we're refactoring this code anyway, can we move the file open/read JSON logic out of the constructor? That code seems like it belongs in the Factory class

Reply:
Let me improve it in the future. In this PR, I just decouple the different wallets from each other.